src/benchmarklib/benchmark_sql_executor.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "sql/sql_pipeline.hpp"
+#include "utils/sqlite_wrapper.hpp"
+
+namespace opossum {
+
+class TransactionContext;
+
+// This class provides SQL functionality to BenchmarkItemRunners. See AbstractBenchmarkItemRunner:_execute_item.
+// If sqlite_wrapper is set, all SQL queries executed in the benchmark will also be executed using SQLite and then
+// validated. For now, we expect items to use a single transaction, which is why the BenchmarkSQLExecutor executes
+// all queries in the same context.
+class BenchmarkSQLExecutor {
+ public:
+  // @param visualize_prefix    Prefix for the filename of the generated query plans (e.g., "TPC-H_6-")
+  BenchmarkSQLExecutor(bool enable_jit, const std::shared_ptr<SQLiteWrapper>& sqlite_wrapper,
+                       const std::optional<std::string>& visualize_prefix);
+
+  // This executes the given SQL query and returns a single table. If visualization and/or verification are enabled,
```

Comment:
Added "records its metrics and returns a single table (the same as SQLPipeline::get_result_table() would)."